python/cudf/cudf/core/column/lists.py

```diff
@@ -69,6 +69,9 @@ def to_arrow(self):
             pa_type, len(self), buffers, children=[elements]
         )
 
+    def __sizeof__(self):
+        return sum(child.__sizeof__() for child in self.base_children)
```

Comment:
Instead of implementing this in the `ListColumn` I think we'd be better off adding the children column sizes in the general Column case: https://github.com/rapidsai/cudf/blob/branch-0.17/python/cudf/cudf/core/column/column.py#L181-L185

Reply:
I agree with you that we should be implementing in the base class - but getting working required making some other fixes too.

```ColumnBase.__sizeof__```  calls ```self.data.size``` which was failing with the ```AttributeError: 'ListDtype' object has no attribute 'itemsize'``` for list columns when trying to get the data property here: https://github.com/rapidsai/cudf/blob/38d25c86592e985d39e6e66d38517016234b1147/python/cudf/cudf/_lib/column.pyx#L113

I've changed in the most recent commit to do add the children sizes in ColumnBase - but I also had to change ListColumn's not to error out on trying to access the ```.data``` property to make this work (in the same way as strings columns, by ensuring that .base_data is null).